python/paddle/fluid/initializer.py

```diff
@@ -482,14 +482,28 @@ def __call__(self, var, block):
         if self._seed == 0:
             self._seed = block.program.random_seed
 
+# to be compatible of fp16 initalizers
+        if var.dtype == VarDesc.VarType.FP16:
+            out_dtype = VarDesc.VarType.FP32
+            out_var = block.create_var(
+                name=unique_name.generate(".".join(
+                    ['truncated_gaussian_random', 'tmp'])),
```

Comment:
Maybe  use `var.name` as part of prefix for `unique_name.generate` a new name.

Reply:
Done